src/dist/block_service/fds/fds_service.cpp

```diff
@@ -484,82 +496,134 @@ fds_file_object::fds_file_object(fds_service *s,
 
 fds_file_object::~fds_file_object() {}
 
-dsn::error_code fds_file_object::get_content(uint64_t pos,
-                                             int64_t length,
-                                             /*out*/ std::ostream &os,
-                                             /*out*/ uint64_t &transfered_bytes)
+error_code fds_file_object::get_file_meta()
 {
-    dsn::error_code err;
+    galaxy::fds::GalaxyFDSClient *c = _service->get_client();
+    try {
+        auto meta = c->getObjectMetadata(_service->get_bucket_name(), _fds_path)->metadata();
+        auto iter = meta.find(fds_service::FILE_LENGTH_CUSTOM_KEY);
+        if (iter != meta.end()) {
```

Comment:
Old code has `dassert(iter != meta.end()`

Reply:
I don't think we should make the server core dump when we encounter some error in fds, so I delete the assert here.

Reply:
If not core dump, you should make sure the following logic is valid. Here, when meta doesn't contain `FILE_LENGTH_CUSTOM_KEY ` or `FILE_MD5_KEY `, `_size ` or `_md5sum ` will not be initialized normally.
In most cases, `dassert` (or similar operation) is used to reduce various of defensive program code, and it is helpful to expose problem as early as possible.